synapse/http/matrixfederationclient.py

```diff
@@ -538,10 +538,10 @@ async def _send_request(
             logger.exception(f"Invalid destination: {request.destination}.")
             raise FederationDeniedError(request.destination)
 
-        if timeout:
-            _sec_timeout = timeout / 1000
-        else:
-            _sec_timeout = self.default_timeout
+        if timeout is None:
+            timeout = int(self.default_timeout)
+
+        _sec_timeout = timeout / 1000
```

Comment:
which means that this is in kiloseconds if `timeout is None`, but seconds otherwise. That seems wrong.

Ditto below.